File: exercises/python_playwright/AJAX_Data/ajax_data_playwright.py
from playwright.sync_api import Page

class AjaxData:
    def __init__(self, page: Page) -> None:
        self.page = page
        self.ajax_button = page.get_by_role("button", name="Button")
        self.ajax_status = page.locator(".bg-success")

    def open(self) -> None:
        self.page.goto("http://uitestingplayground.com/ajax")

    def click_ajax_button(self) -> None:
        self.ajax_button.click()

File: exercises/javascript_playwright/ajax-data/ajaxDataPlaywright.js
export class AJAXDataPage {
    /**
     *
     * @param {import('@playwright/test').Page} page
     */
    constructor(page) {
        this.page = page;
        this.AJAXButton = page.locator("#ajaxButton");
        this.successMessage = page.locator(".bg-success");
    }
    async openPage() {
        await this.page.goto("http://uitestingplayground.com/ajax");
    }

    async clickAJAXButton() {
        await this.AJAXButton.click();
    }
}
